enhancements/20230504-freeze-fs-during-snapshot-creation.md

```diff
@@ -0,0 +1,69 @@
+# Freeze FS during snapshot creation
+
+## Summary
+
+HEP introduces proposed changes needed to harvester to meet the requirements of performing a fs freeze on guests when a backup is performed.
+
+Currently when a user initiates a VM backup, no guest level action is performed to ensure the filesystems are sync or io is paused.
+
+If a backup is triggered during a high IO event, the backup can be inconsistent.
+
+### Related Issues
+
+[GH issue 1723 ](https://github.com/harvester/harvester/issues/1723)
+
+### Goals
+
+- Attempt to perform fs freeze if qemu-guest-agent is installed in the guest when a backup is triggered
+
+## Proposal
+
+### User Stories
+
+#### Need for consistent backups for VM workloads
+#####  User Experience In Detail
+A VM backup initiated during times of high IO, can be inconsistent. This can be due to the fact the backup controller doesn't attempt to pause fs io when a backup is performed.
+
+### API changes
+No harvester api changes needed. The only changes will be in the backup controller logic.
+
+## Design
+
+### Implementation Overview
+
+The changes need to be implemented in the harvester [backup controller](https://github.com/harvester/harvester/blob/master/pkg/controller/master/backup/backup.go)
+
+Additional logic will be needed to check the following:
+* check if qemu-guest-agent is available, this will be available via vmi status by checking if condition for GuestAgent connected is true
+```yaml
+  conditions:
+  - lastProbeTime: null
+    lastTransitionTime: "2023-05-04T00:41:03Z"
+    status: "True"
+    type: Ready
+  - lastProbeTime: null
+    lastTransitionTime: null
+    status: "True"
+    type: LiveMigratable
+  - lastProbeTime: "2023-05-04T00:41:23Z"
+    lastTransitionTime: null
+    status: "True"
+    type: AgentConnected
+```
+
+* if guest agent is present, then [reconcileVolumeSnapshots](https://github.com/harvester/harvester/blob/master/pkg/controller/master/backup/backup.go#L443) could trigger a fs freeze on the VM using [freeze/unfreeze subresource api](https://github.com/kubevirt/kubevirt/blob/main/docs/freeze.md) with a small unfreezeTimeout.
+
+* volume backups proceed as expected.
+### Test plan
```

Comment:
can we add an additional case to verify the difference before and after adding the fs freeze enhancement?